cert/aziot-certd-config/src/lib.rs

```diff
@@ -247,6 +251,23 @@ impl std::convert::TryFrom<&CertSubject> for openssl::x509::X509Name {
     }
 }
 
+pub fn deserialize_map<'de, D>(de: D) -> Result<BTreeMap<String, Url>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let map: BTreeMap<String, Url> = BTreeMap::deserialize(de)?;
+    for url in map.values() {
+        if url.scheme() == "http" {
+            println!(
+                "Warning: EST server URL {} is configured with unencrypted HTTP, which may expose device to man-in-the-middle attacks.
+                    To clear this warning, configure HTTPS for your EST server and update the URL.",
+                url.as_str()
+            );
```

Comment:
```suggestion
            eprintln!(
                "Warning: EST server URL {:?} is configured with unencrypted HTTP, which may expose device to man-in-the-middle attacks. To clear this warning, configure HTTPS for your EST server and update the URL.", url.as_str()
            );
```
Rust strings do not do whitespace prefix normalization. `eprintln` is to make it clearer that the warning is not a normal result. The `{:?}` format string uses debug formatting, which in the case of `&str` wraps the string in double quotes.

Reply:
Thank you for the input. Addressed this comment as well as others